live-hls.js

```diff
@@ -370,18 +380,25 @@ createManifest = function(mfest, duration, event) {
  * @return {string} the lines of the playlist that would be available
  * at the specified time
  */
-filterPlaylist = function(playlist, time, options) {
+
+filterPlaylist = function(manifest, playlist, time, options) {
+
   var startposition;
+  var filteredLines;
   var endposition;
-  var overflow = 0;
+  var header = manifest.header;
+  var resource = manifest.resources;
+  var overflow=0;
   var lines = playlist.split('\n');
 
   options.init=getInitialValue(lines);
   // the number of the time-varying lines to be shown
   var temptime=(time*0.001);
+  var programDateTime = lines[options.init];
```

Comment:
This assumes both that there will be an `EXT-X-PROGRAM-DATE-TIME` and that the `EXT-X-PROGRAM-DATE-TIME` tag will be the last tag before the `EXTINF` but this is not always going to be the case. We want to support simulating streams with this tag, but not do anything different and continue working for streams that do not have the tag